vocode/streaming/agent/state_agent.py

```diff
@@ -651,12 +662,17 @@ def update_history(
                 )
             )
         else:
-            self.json_transcript.entries.append(
-                StateAgentTranscriptMessage(role=role, message=message)
+
+            transcript_entry = StateAgentTranscriptMessage(
+                role=role, message=message, message_sent=""
             )
+            if len(messages_to_add) > 0 and not speak:
+                self.json_transcript.entries.extend(messages_to_add)
+            else:
+                self.json_transcript.entries.append(transcript_entry)
 
+        self.logger.info(f"{role=}, {message=}, {speak=}, {transcript_entry=}")
```

Comment:
If there are messages to add. add them instead of creating a `transcript_entry`. This should only ever run when speak is false. Should probably add a warning to the logs if this is ever not the case.